File: Backend/Challenge.Domain/Aggregates/PokemonMove.cs
using Challenge.Domain.Interfaces.Aggregates;
using System.Text.Json.Serialization;

namespace Challenge.Domain.Aggregates
{
    public class PokemonMove : IMove
    {
        public PokemonMove()
        {
        }
        [JsonPropertyName("version_group_details")]
        public List<VersionGroupDetail> VersionGroupDetails { get; set; } = new List<VersionGroupDetail>();

        public MoveDescription Move { get; set; } = new MoveDescription();  

    }
}

File: JsonFiles/api/models/PokemonMove.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { MoveDescription } from './MoveDescription';
import type { VersionGroupDetail } from './VersionGroupDetail';

export type PokemonMove = {
    version_group_details?: Array<VersionGroupDetail> | null;
    move?: MoveDescription;
};
